framework/helpers/BaseStringHelper.php

```diff
@@ -104,12 +104,15 @@ public static function dirname($path)
      */
     public static function truncate($string, $length, $suffix = '...', $encoding = null, $asHtml = false)
     {
+        if ($encoding === null) {
+            $encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
```

Comment:
>  harder to read

```php
$encoding = Yii::$app ? Yii::$app->charset : 'UTF-8';
// vs
$encoding = self::getEncoding();
```

I think the second case is easier to read.

> degrading performance

Microseconds degrading performance. I don't thinkthats an argument.

> Even though it is happening often that still does not justify an extra method call

If you want to stay DRY yes thats a great justification.

and that method can be used on other places such as widgets or views.